### PR TITLE
feat(rbac): save role modification information to the metadata

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this pull request do

Save role modification information to the metadata:

- author
- modifiedBy
- createdAt
- lastModified

In the UI will be used : lastModified and lastModified

### What issue is referenced in this pull request?

https://issues.redhat.com/browse/RHIDP-1382
https://github.com/janus-idp/backstage-plugins/issues/900

### Screenshot

screenshot-localhost_3000-2024 04 05-16_48_09
](url)

### CODE REVIEW

File: plugins/rbac/src/components/RoleOverview/AboutCard.tsx [hunk 1]
```diff
@@ -53,6 +53,23 @@ export const AboutCard = ({ roleName }: AboutCardProps) => {
   if (loading) {
     return <Progress />;
   }
+
+  let lastModified = role?.metadata?.lastModified;
+  if (lastModified) {
```
Comment (invincibleJai) [inline plugins/rbac/src/components/RoleOverview/AboutCard.tsx:58]: would be good to add unit test for this.

**Comment by invincibleJai** [comment]
Thanks @AndrienkoAleksandr , Frontend plugin changes looks good to me

cc @PatAKnight

**Review by PatAKnight — COMMENTED** [review]
Looks good so far, just had a  couple of small issues while testing.

One thing that I noticed is that changes to the CSV file while it is running will not update the `lastModified` field.

Some scenarios that I tried are:
Adding, editing, removing roles and permissions in REST and CSV
Adding, editing, removing description from REST

I was able to swap between a couple of users to test how it looks.

**Review by PatAKnight — APPROVED** [review]
Retested after the changes and everything looks good
/lgtm

**Pull request merged**